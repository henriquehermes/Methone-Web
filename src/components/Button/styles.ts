import styled, { keyframes } from "styled-components";

const fadeIn = keyframes`
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;

export const Container = styled.button`
	display: flex;
	width: 100%;
	height: 50px;
	border-radius: 6px;
	background-color: ${props => props.theme.royalblue};
	justify-content: center;
	align-items: center;
	cursor: pointer;
	z-index: 99;
	animation: 1.5s ${fadeIn} forwards;
	opacity: 0;

	transition: transform 0.2s;

	&:hover {
		transform: scale(1.05);
	}
`;

export const ButtonText = styled.h1`
	font-size: 20px;
	font-family: "Montserrat" sans-serif;
	color: ${props => props.theme.white};
	width: 100%;
	text-align: center;
`;

export const Loader = styled.div`
	border: 2px solid ${props => props.theme.white};
	border-top: 2px solid ${props => props.theme.royalblue};
	border-radius: 50%;
	width: 25px;
	height: 25px;
	animation: spin 1s linear infinite;

	@keyframes spin {
		0% {
			transform: rotate(0deg);
		}
		100% {
			transform: rotate(360deg);
		}
	}
`;
