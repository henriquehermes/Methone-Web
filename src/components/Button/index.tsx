import React from 'react';
import { useSelector } from 'react-redux';

import { Container, ButtonText, Loader } from './styles';

interface Props {
    label: string;
    onAction?: any;
}

interface General {
    isLoading: boolean;
}

interface RootState {
    general: General;
}

const Button: React.FC<Props> = ({ label, onAction }) => {
    const isLoading = useSelector((state: RootState) => state.general.isLoading);

    return (
        <Container
            disabled={isLoading}
            onClick={() => (onAction ? onAction() : null)}
        >
            {isLoading ? <Loader /> : <ButtonText>{label}</ButtonText>}
        </Container>
    );
};

export default Button;
