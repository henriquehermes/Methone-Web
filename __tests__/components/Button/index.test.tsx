import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";

import Button from "../../../src/components/Button";
import { Provider } from "react-redux";

const mockStore = configureStore([]);

const store = mockStore({
	general: {
		isLoading: false,
	},
});

const storeLoader = mockStore({
	general: {
		isLoading: true,
	},
});

interface Props {
    label: string;
    onAction?: any;
}

const newWrapper = ({ label, onAction }: Props) => (
	<Provider store={store}>
		<Button label={label} onAction={onAction} />
	</Provider>
);

const wrapperLoader = ({ label, onAction }: Props) => (
	<Provider store={storeLoader}>
		<Button label={label} onAction={onAction} />
	</Provider>
);

const goToQuestions = jest.fn()

describe("components - Button", () => {
	it("renders correctly", () => {
		const tree = renderer.create(newWrapper({ label: "Label", onAction: goToQuestions })).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("renders with loading true", () => {
		const tree = renderer.create(wrapperLoader({ label: "Label", onAction: {} })).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
