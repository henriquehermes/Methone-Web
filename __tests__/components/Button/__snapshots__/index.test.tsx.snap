// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Button renders correctly 1`] = `
<button
  className="sc-AxjAm hBJLha"
  disabled={false}
  onClick={[Function]}
>
  <h1
    className="sc-AxirZ sYFMH"
  >
    Label
  </h1>
</button>
`;

exports[`components - Button renders with loading true 1`] = `
<button
  className="sc-AxjAm hBJLha"
  disabled={true}
  onClick={[Function]}
>
  <div
    className="sc-AxiKw DDEvM"
  />
</button>
`;
